### PR TITLE
Support OSD file backing

### PR DESCRIPTION
**Pull request by sabaini**
Add support for auto-creating loopback backing for OSDs

### CODE REVIEW

File: microceph/ceph/osd.go [hunk 1]
```diff
@@ -292,22 +295,164 @@ func setStablePath(storage *api.ResourcesStorage, param *types.DiskParameter) er
...
 }
 
+// parseBackingSpec parses a loopback file specification.
+// The specification is of the form "loop,<size><unit>,<number>".
+// The function returns the size in MB and the number of disks.
+func parseBackingSpec(spec string) (uint64, int, error) {
+	r := regexp.MustCompile("loop,([1-9][0-9]*[MGT]),([1-9][0-9]*)")
+
+	match := r.FindStringSubmatch(spec)
+	if match != nil {
```
Comment (UtkarshBhatthere) [inline microceph/ceph/osd.go:305]: This is slightly different from the function structure which is (usually)"
```
if ERR {
    # RETURN ERROR
}

# PROCEED
```

but rather:
```
if SUCCESS {
    # Core Logic
}
# Return Error
```

The only nit being the structure is unconventional for quick reading.

**Review by UtkarshBhatthere — CHANGES_REQUESTED** [review]
Did a very quick review, LGTM in general, just a small nit and the CLI changes seem missing ?

**Comment by sabaini** [comment]
> Did a very quick review, LGTM in general, just a small nit and the CLI changes seem missing ?

Thanks! I've now added longer help text for clarity

File: microceph/ceph/osd.go [hunk 2]
```diff
@@ -292,22 +295,163 @@ func setStablePath(storage *api.ResourcesStorage, param *types.DiskParameter) er
...
+	sizeStr := match[1][:len(match[1])-1]
+	unit := match[1][len(match[1])-1:]
+
+	size, err := strconv.ParseUint(sizeStr, 10, 64)
+	if err != nil {
+		return 0, 0, fmt.Errorf("failed to parse size from spec %s: %w", spec, err)
+	}
+
+	// Convert the size to MB.
+	switch strings.ToUpper(unit) {
```
[the remark this answers is not in the document]
Comment (sabaini) [reply microceph/ceph/osd.go:318]: The regexp should take care of that -- it matches only M, G or T. Anything else won't match and cause an "illegal spec" error

File: microceph/ceph/osd.go [hunk 3]
```diff
@@ -292,22 +295,163 @@ func setStablePath(storage *api.ResourcesStorage, param *types.DiskParameter) er
...
+}
+
 // AddOSD adds an OSD to the cluster, given the data, WAL and DB devices and their respective
 // flags for wiping and encrypting.
 func AddOSD(s *state.State, data types.DiskParameter, wal *types.DiskParameter, db *types.DiskParameter) error {
 	logger.Debugf("Adding OSD %s", data.Path)
 
+	var err error
+
+	// sanity: loopback file and WAL/DB are mutually exclusive
```
[the remark this answers is not in the document]
Comment (sabaini) [reply microceph/ceph/osd.go:435]: My reasoning was that having loop files and using separate WAL/DB devices are orthogonal use cases, and we'd complicate implementation but also user interface surface without much benefit

**Pull request merged**